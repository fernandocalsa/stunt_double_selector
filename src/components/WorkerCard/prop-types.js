import PropTypes from 'prop-types'

export default {
  worker: PropTypes.shape({
    imageUrl: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired
  }).isRequired,
  onHandleWorker: PropTypes.func.isRequired
}
