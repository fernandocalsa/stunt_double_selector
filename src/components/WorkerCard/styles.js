const icon = {
  width: 50,
  height: 50
}

export default {
  buttons: {
    display: 'flex',
    justifyContent: 'space-between',
    margin: '10px 0'
  },
  iconApprove: {
    ...icon,
    color: 'green'
  },
  iconReject: {
    ...icon,
    color: 'red'
  }
}
