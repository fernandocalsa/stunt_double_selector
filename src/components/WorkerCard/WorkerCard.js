import React, { Component } from 'react'
import { FaCheckCircle, FaTimesCircle } from 'react-icons/fa'
import Draggable from 'react-draggable'
import WorkerInfo from '../WorkerInfo'
import styles from './styles'

class WorkerCard extends Component {
  handleStop = (e, data) => {
    if (data.lastX < -100) {
      console.log('reject')
    } else if (data.lastX > 100) {
      console.log('approve')
    }
  }
  render() {
    return (
      <div style={{
        width: 500
      }}>
        <Draggable
          axis="x"
          defaultPosition={{x: 0, y: 0}}
          onStop={this.handleStop}
          position={{
            x: 0,
            y: 0
          }}
        >
          <div>
          <WorkerInfo
            imageUrl="https://images.pexels.com/photos/614810/pexels-photo-614810.jpeg"
            firstName="Fer"
            lastName="Calvo"
          />
          </div>
        </Draggable>
        <div style={styles.buttons}>
          <FaCheckCircle style={styles.iconApprove}/>
          <FaTimesCircle style={styles.iconReject} />
        </div>
      </div>
    )
  }
}

export default WorkerCard
