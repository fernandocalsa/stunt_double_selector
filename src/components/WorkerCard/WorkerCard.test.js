import React from 'react'
import { shallow } from 'enzyme'
import WorkerCard from './WorkerCard'

describe('WorkerCard', () => {
  it('Should render the worker info and buttons', () => {
    const props = {
      worker: {
        imageUrl: "test.jpg",
        name: "Charlie"
      },
      onHandleWorker: jest.fn()
    }
    const output = shallow(<WorkerCard {...props} />)
    expect(output).toMatchSnapshot()
  })

  it('card moves more than 100px to the right then it approves the worker', () => {
    const props = {
      worker: {
        imageUrl: "test.jpg",
        name: "Charlie"
      },
      onHandleWorker: jest.fn()
    }
    const output = shallow(<WorkerCard {...props} />)
    output.instance().handleStop(null, {lastX: 101})
    expect(props.onHandleWorker).toHaveBeenCalledWith(true)
  })

  it('card moves more than 100px to the left then it rejects the worker', () => {
    const props = {
      worker: {
        imageUrl: "test.jpg",
        name: "Charlie"
      },
      onHandleWorker: jest.fn()
    }
    const output = shallow(<WorkerCard {...props} />)
    output.instance().handleStop(null, {lastX: -101})
    expect(props.onHandleWorker).toHaveBeenCalledWith(false)
  })

  it('card moves less than 100px then onHandleWorker is not called', () => {
    const props = {
      worker: {
        imageUrl: "test.jpg",
        name: "Charlie"
      },
      onHandleWorker: jest.fn()
    }
    const output = shallow(<WorkerCard {...props} />)
    output.instance().handleStop(null, {lastX: 99})
    expect(props.onHandleWorker).not.toHaveBeenCalled()
  })

  it('clicks on approve icon approves the worker', () => {
    const props = {
      worker: {
        imageUrl: "test.jpg",
        name: "Charlie"
      },
      onHandleWorker: jest.fn()
    }
    const output = shallow(<WorkerCard {...props} />)
    output.find('#approveBtn').simulate('click')
    expect(props.onHandleWorker).toHaveBeenCalledWith(true)
  })

  it('clicks on reject icon rejects the worker', () => {
    const props = {
      worker: {
        imageUrl: "test.jpg",
        name: "Charlie"
      },
      onHandleWorker: jest.fn()
    }
    const output = shallow(<WorkerCard {...props} />)
    output.find('#rejectBtn').simulate('click')
    expect(props.onHandleWorker).toHaveBeenCalledWith(false)
  })
})