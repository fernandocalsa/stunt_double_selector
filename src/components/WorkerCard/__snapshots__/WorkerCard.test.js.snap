// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorkerCard Should render the worker info and buttons 1`] = `
<div
  style={
    Object {
      "width": 500,
    }
  }
>
  <Draggable
    allowAnyClick={false}
    axis="x"
    bounds={false}
    cancel={null}
    defaultClassName="react-draggable"
    defaultClassNameDragged="react-draggable-dragged"
    defaultClassNameDragging="react-draggable-dragging"
    defaultPosition={
      Object {
        "x": 0,
        "y": 0,
      }
    }
    disabled={false}
    enableUserSelectHack={true}
    grid={null}
    handle={null}
    offsetParent={null}
    onDrag={[Function]}
    onMouseDown={[Function]}
    onStart={[Function]}
    onStop={[Function]}
    position={
      Object {
        "x": 0,
        "y": 0,
      }
    }
    transform={null}
  >
    <div>
      <WorkerInfo
        imageUrl="test.jpg"
        name="Charlie"
      />
    </div>
  </Draggable>
  <div
    style={
      Object {
        "display": "flex",
        "justifyContent": "space-between",
        "margin": "10px 0",
      }
    }
  >
    <FaCheckCircle
      id="approveBtn"
      onClick={[Function]}
      style={
        Object {
          "color": "green",
          "height": 50,
          "width": 50,
        }
      }
    />
    <FaTimesCircle
      id="rejectBtn"
      onClick={[Function]}
      style={
        Object {
          "color": "red",
          "height": 50,
          "width": 50,
        }
      }
    />
  </div>
</div>
`;
